invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Document/Forms/DocumentEditor/components/DocumentForm/DocumentForm.js

```diff
@@ -0,0 +1,68 @@
+// EditUserDialog.js
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getIn } from 'formik';
+import _ from 'lodash';
+import { BaseForm, StringField } from '../../../../../../../forms';
+import { document as documentApi } from '../../../../../../../common/api/documents/document';
+import { BackOfficeRoutes } from '../../../../../../../routes/urls';
+import { goTo } from '../../../../../../../history';
+import { TagsField } from './components';
+
+export class DocumentForm extends Component {
+  constructor(props) {
+    super(props);
+    this.formInitialData = props.data;
+    this.successSubmitMessage = props.successSubmitMessage;
+    this.title = props.title;
+    this.pid = props.pid;
+  }
+
+  updateDocument = (pid, data) => {
+    return documentApi.update(pid, data);
+  };
+
+  createDocument = data => {
+    return documentApi.create(data);
+  };
+
+  successCallback = response => {
+    goTo(
+      BackOfficeRoutes.documentDetailsFor(getIn(response, 'data.metadata.pid'))
+    );
+  };
+
+  render() {
+    return (
+      <BaseForm
+        initialValues={
+          this.formInitialData ? this.formInitialData.metadata : {}
+        }
+        editApiMethod={this.updateDocument}
+        createApiMethod={this.createDocument}
+        successCallback={this.successCallback}
+        successSubmitMessage={this.successSubmitMessage}
+        title={this.title}
+        pid={this.pid ? this.pid : undefined}
+        submitSerializer={values => {
+          const submittingValues = { ...values };
+          submittingValues['tag_pids'] = submittingValues['tags'].map(
+            tag => tag.pid
+          );
+          delete submittingValues['tags'];
```

Comment:
shouldn't this be handled by the loader on the backend?

Reply:
So I had this doubt, should I pass everything and loader does what it should be or manipulate it before. I was wondering if you want to see the payload that goes through in your browser you will not see the correct one that is actually committed in the backend. @ntarocco WDYT?